samza-core/src/main/java/org/apache/samza/task/RunLoop.java

```diff
@@ -457,17 +457,25 @@ private void process() {
       final IncomingMessageEnvelope envelope = state.fetchEnvelope();
       log.trace("Process ssp {} offset {}", envelope.getSystemStreamPartition(), envelope.getOffset());
 
-      final ReadableCoordinator coordinator = new ReadableCoordinator(task.taskName());
-      TaskCallbackFactory callbackFactory = new TaskCallbackFactory() {
-        @Override
-        public TaskCallback createCallback() {
-          state.startProcess();
-          containerMetrics.processes().inc();
-          return callbackManager.createCallback(task.taskName(), envelope, coordinator);
-        }
+      Runnable processWorker = () -> {
+        final ReadableCoordinator coordinator = new ReadableCoordinator(task.taskName());
```

Comment:
Just curious. 

Previously, we used to have in-order delivery of messages to the task implementation(where max-in-flight messages controlled by `task.max.concurrency` configuration). This behavior was possible since the delivery/dispatch of messages to task implementation would happen only from the event-loop. 

Now, we enqueue process-tasks into the `ExecutorService` from event-loop. Tasks that are submitted to `ExecutorService` can be executed in any order(Interface contract does not guarantee in-order execution of submitted tasks). This could potentially deliver/dispatch messages out of order to the task implementation(i.e message with offset 2 can be delivered before message with offset 1). 

Can you please confirm if the above message-dispatch behavior to task is plausible?

Reply:
Not applicable anymore